# Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd

In [4]:
data = {'Student_names' : ['Sumit', 'Prateek' , 'Ashish' , 'Kavya' , 'Jatin'],
       'Time_Spend' : [8,6,5,3,1],
       'Scores' : [430,400,380,330,230]}

df = pd.DataFrame(data)
df

Student_names  Time_Spend  Scores
0         Sumit           8     430
1       Prateek           6     400
2        Ashish           5     380
3         Kavya           3     330
4         Jatin           1     230

In [6]:
correlation = df.corr(method='pearson')
correlation

/tmp/ipykernel_84/4041265564.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlation = df.corr(method='pearson')


Time_Spend    Scores
Time_Spend    1.000000  0.966722
Scores        0.966722  1.000000

Hence, we can interpret that their is positive ccorrelation of 0.966722 between the amount of scores with respect to the time of studying.

# Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [13]:
emp_data = {'Employees_names' : ['Sumit', 'Prateek' , 'Ashish' , 'Kavya' , 'Jatin'],
            'Sleep_Time' : [7,6,8,5,7],
           'Satisfaction_level' : [5,4,8,4,7]}

emp_df = pd.DataFrame(emp_data)
print(emp_df)

correlation_of_emp_data =emp_df.corr(method='spearman')
correlation_of_emp_data

  Employees_names  Sleep_Time  Satisfaction_level
0           Sumit           7                   5
1         Prateek           6                   4
2          Ashish           8                   8
3           Kavya           5                   4
4           Jatin           7                   7


/tmp/ipykernel_84/2581173567.py:8: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlation_of_emp_data =emp_df.corr(method='spearman')


Sleep_Time  Satisfaction_level
Sleep_Time            1.000000            0.947368
Satisfaction_level    0.947368            1.000000

Hence, we can interpret that their is positive ccorrelation of 0.947368 between the Satisfaction_level with respect to the Sleep_Time .

# Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [14]:
import numpy as np
import pandas as pd

In [50]:
hours_of_exercise = np.random.randint(5,28,50)
body_mass_index = np.random.randint(19,24,50)   

hours_of_exercise  body_mass_index
0                 11               22
1                  6               21
2                 19               21
3                 25               21
4                 24               23

In [52]:
health_data = pd.DataFrame({'hours_of_exercise' : hours_of_exercise, 
                            'body_mass_index': body_mass_index})
health_data.head()

hours_of_exercise  body_mass_index
0                 11               22
1                  6               21
2                 19               21
3                 25               21
4                 24               23

In [53]:
# Pearson correlation coefficient

p_correlation_health_data = health_data.corr(method='pearson')
(p_correlation_health_data)

# Hence, we can interpret that their is positive correlation of 0.122547 between the hours_of_exercise and body_mass_index using Pearson correlation coefficient.

hours_of_exercise  body_mass_index
hours_of_exercise           1.000000         0.122547
body_mass_index             0.122547         1.000000

In [44]:
# Spearman's rank correlation

s_correlation_health_data = health_data.corr(method='spearman')
s_correlation_health_data

# Hence, we can interpret that their is positive correlation of 0.076499 between the hours_of_exercise and body_mass_index using Spearman's rank correlation.

hours_of_exercise  body_mass_index
hours_of_exercise           1.000000         0.076499
body_mass_index             0.076499         1.000000

# Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [32]:
television_hours = np.random.randint(1,4,51)
physical_activity_level = np.random.randint(1,5,51)   


activity_relationship = pd.DataFrame({'television_hours' : television_hours, 
                            'physical_activity_level': physical_activity_level})

p_correlation_activity_relationship = activity_relationship.corr(method='pearson')
(p_correlation_activity_relationship)

# Hence, we can interpret that their is positive correlation of 0.081633 between the television_hours and physical_activity_level using Pearson correlation coefficient.

television_hours  physical_activity_level
television_hours                 1.000000                 0.081633
physical_activity_level          0.081633                 1.000000

# Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [40]:
survey_data = pd.DataFrame({'Age(Years)' : [25,42,37,19,31,28],
             'Soft_drink_Preference' : ['Coke','Pepsi','Mountain dew','Coke','Pepsi','Coke']})

survey_data

## I feel like the question is incomplete or I may be wrong in detecting what the question is asking.

Age(Years) Soft_drink_Preference
0          25                  Coke
1          42                 Pepsi
2          37          Mountain dew
3          19                  Coke
4          31                 Pepsi
5          28                  Coke

# Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [54]:
sales_calls = np.random.randint(30,60,30)
sales_made_per_week = np.random.randint(50,150,30)   


In [55]:
company_data = pd.DataFrame({'sales_calls' : sales_calls, 
                            'sales_made_per_week': sales_made_per_week})

p_correlation_company_data = company_data.corr(method='pearson')
(p_correlation_company_data)

# Hence, we can interpret that their is positive correlation of 0.026984 between the sales_calls and sales_made_per_week using Pearson correlation coefficient.

sales_calls  sales_made_per_week
sales_calls             1.000000             0.026984
sales_made_per_week     0.026984             1.000000